### PR TITLE
feat: enabled customization of project audit logs retention period

### PR DESCRIPTION
**Pull request by sheensantoscapadngan**
# Description 📣
- This PR will enable users to configure audit logs retention period per project. For org-level logs, retention is set to their plan's value

image

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: backend/src/server/routes/v1/project-router.ts [hunk 1]
```diff
@@ -372,6 +372,44 @@ export const registerProjectRouter = async (server: FastifyZodProvider) => {
...
+    url: "/:workspaceSlug/audit-logs-retention",
+    config: {
+      rateLimit: writeLimit
+    },
+    schema: {
+      params: z.object({
+        workspaceSlug: z.string().trim()
+      }),
+      body: z.object({
+        auditLogsRetentionDays: z.number().min(1).max(100)
```
Comment (akhilmhdh) [inline backend/src/server/routes/v1/project-router.ts:386]: Let keep the max even smaller. 100 is more than 3 months a bit too much. 

@maidul98 Your opinion please?

File: backend/src/services/project/project-service.ts [hunk 2]
```diff
@@ -446,6 +447,43 @@ export const projectServiceFactory = ({
...
+    actorId,
+    actorOrgId,
+    actorAuthMethod,
+    auditLogsRetentionDays,
+    workspaceSlug
+  }: TUpdateAuditLogsRetentionDTO) => {
+    const project = await projectDAL.findProjectBySlug(workspaceSlug, actorOrgId);
+    if (!project) {
+      throw new NotFoundError({
+        message: "Project not found."
```
Comment (akhilmhdh) [inline backend/src/services/project/project-service.ts:461]: Good job using `NotFoundError`. Good API convention

File: frontend/src/views/Settings/ProjectSettingsPage/components/AuditLogsRetentionSection/AuditLogsRetentionSection.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+import { createNotification } from "@app/components/notifications";
+import { Button, FormControl, Input, UpgradePlanModal } from "@app/components/v2";
+import { useProjectPermission, useSubscription, useWorkspace } from "@app/context";
+import { usePopUp } from "@app/hooks";
+import { ProjectMembershipRole } from "@app/hooks/api/roles/types";
+import { useUpdateWorkspaceAuditLogsRetention } from "@app/hooks/api/workspace/queries";
+
+const formSchema = z.object({
+  auditLogsRetentionDays: z.coerce.number().min(0)
```
Comment (akhilmhdh) [inline frontend/src/views/Settings/ProjectSettingsPage/components/AuditLogsRetentionSection/AuditLogsRetentionSection.tsx:13]: Please add `max` here also

**Review by akhilmhdh — CHANGES_REQUESTED** [review]

File: backend/src/server/routes/v1/project-router.ts (hunk 1 above)
Comment (sheensantoscapadngan) [reply backend/src/server/routes/v1/project-router.ts:386]: actually, the max is dictated by the value of `auditLogsRetentionDays` in the project's subscriptio plan. We're also doing this check in the backend. What do you think? 
image

Comment (sheensantoscapadngan) [reply backend/src/server/routes/v1/project-router.ts:386]: i'm thinking some organizations might need more days, so to be flexible I didn't define a max in the schema but rather did a check against the subscription in the service layer instead

Comment (maidul98) [reply backend/src/server/routes/v1/project-router.ts:386]: Retention period to pricing is a seperate topic, we can open up a thread about it in slack @akhilmhdh . I think this setting should only really be shown to self hosted instances. On cloud, all the user could do is lower the retention days which i don't think they care to do

File: backend/src/services/project/project-service.ts (hunk 2 above)
Comment (maidul98) [reply backend/src/services/project/project-service.ts:461]: nice! we should start to use this in other places too

**Review by akhilmhdh — APPROVED** [review]

**Review by akhilmhdh — APPROVED** [review]

**Pull request merged**